src/loopy/loopy.c

```diff
@@ -0,0 +1,143 @@
+/* Copyright (c) 2015 CoreOS. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+
+#include <sys/mount.h>
+#include <linux/loop.h>
+#include <sys/file.h>
+#include <libmount/libmount.h>
+#include <unistd.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+static int get_node(const char *image_path, char **loopdev) {
+
+        int fd = -1, nr = -1, loop = -1, control = -1, success = -1;
+
+        fd = open(image_path, O_CLOEXEC|O_RDWR|O_NONBLOCK|O_NOCTTY);
+        if (fd < 0) {
+                perror("Failed to open file descriptor for provided image");
+                goto out;
+        }
+
+        control = open("/dev/loop-control", O_RDWR|O_CLOEXEC|O_NOCTTY|O_NONBLOCK);
+        if (control < 0) {
+                perror("Failed to open /dev/loop-control");
+                goto out;
+        }
+
+        nr = ioctl(control, LOOP_CTL_GET_FREE);
+        if (nr < 0) {
+                perror("Failed to allocate loop device node");
+                goto out;
+        }
+
+         if (asprintf(loopdev, "/dev/loop%i", nr) < 0) {
+                perror("Failed to retrieve path of loop device node");
+                goto out;
+        }
+
+        printf("initializing loop device node at /dev/loop%d \n", nr);
+
+        loop = open(*loopdev, O_CLOEXEC|O_RDWR|O_NONBLOCK|O_NOCTTY);
```

Comment:
Honestly not sure. I just copied this part from nspawn. But let me go double check all these attributes.